### PR TITLE
Make sure the pythonisation doc shows up at the end of the classes doc

### PR DESCRIPTION
**Pull request by couet**
The Python documentation added by the scripts `extract_docstrings.py` and `print_roofit_pyz_doctrings.py` was supposed to appear at the end of the classes' documentations they complete. To make sure it is the case one need to 
list the `.pyzdoc` files individually at the end of the `INPUT` list in Doxyfile. Any other technique appeared to not work as wanted. For example with some of them the markdown documentation present in the pythonisation frame was not interpreted properlly.

# This Pull request:

This issue is fixed by generating the `INPUT` list via a script which lists all the .pyzdoc file at the end of the directory list.
This list is put in a separated file which is included at the right place in Doxyfile.

## Changes or fixes:

  - `Doxyfile`
      - Define `INPUT`via a `@INCLUDE`
  - `extract_docstrings.py`
       - make sure the python doc file have the extension `.pyzdoc` 
  - `print_roofit_pyz_doctrings.py`
      - Change a group definition  
  - `makeinput.sh`
      - New script building the `INPUT` list 
  - `Makefile`
       - Call `makeinput.sh`
       - Create a `pyzdoc` folder where all the `.pyzdoc` file are moved

### CODE REVIEW

File: documentation/doxygen/Doxyfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Doxyfile 1.9.0
+# Doxyfile 1.9.3
```
Comment (etejedor) [inline documentation/doxygen/Doxyfile:1]: This file contains many changes that are not strictly related to the correct placement of the PyROOT boxes (basically everything except for the dynamic `INPUT`, I believe). Could you place those changes in a separate commit?

File: documentation/doxygen/Makefile [hunk 2]
```diff
@@ -40,18 +41,22 @@ images:
 	cp images/* $(DOXYGEN_IMAGE_PATH)
 
 pyzdoc:
+	$(call MkDir,$(DOXYGEN_PYZDOC_PATH))
 	$(PYTHON_EXECUTABLE) extract_docstrings.py $(PYZ_DIR)
 	$(PYTHON_EXECUTABLE) print_roofit_pyz_doctrings.py > $(PYZ_DIR)/_roofit.pyzdoc
+	mv $(PYZ_DIR)/*.pyzdoc $(DOXYGEN_PYZDOC_PATH)
```
Comment (etejedor) [inline documentation/doxygen/Makefile:47]: I think it would be cleaner to pass to `extract_docstrings.py` and `print_roofit_pyz_doctrings.py` (I just noticed the missing `s` in the name of this script btw @guitargeek ) the output path (`$(DOXYGEN_PYZDOC_PATH)`) so they write directly there, instead of generating in `$(PYZ_DIR)` and then moving the results of both.

File: documentation/doxygen/extract_docstrings.py [hunk 3]
```diff
@@ -64,6 +64,6 @@ def run_fast_scandir(dir, ext):    # dir: str, ext: list
     module = ast.parse(file_contents)
     ds = ast.get_docstring(module)
     if ds is not None:
-        with open(pyz_file_path + '.pyzdoc', 'w') as pyz_doc_file:
+        with open(pyz_file_path + 'zdoc', 'w') as pyz_doc_file:
```
Comment (etejedor) [inline documentation/doxygen/extract_docstrings.py:67]: I think this new concatenation deserves at least a comment to explain that the input file name ends in .py and we write the output as `.pyzdoc`.

File: documentation/doxygen/Doxyfile (hunk 1 above)
Comment (couet) [reply documentation/doxygen/Doxyfile]: Those changes were only an upgrade to the new Doxygen version ... but fine I removed them and do only the changes related to the Python Frame.

File: documentation/doxygen/Makefile (hunk 2 above)
Comment (couet) [reply documentation/doxygen/Makefile:47]: I did it that way to avid to modify your code to much guys ...

File: documentation/doxygen/extract_docstrings.py (hunk 3 above)
Comment (couet) [reply documentation/doxygen/extract_docstrings.py:67]: how do you put comments "inline"with python ? the equivalent of:
```
 int i = 1; // I is equal to one
```
in C++ ?

File: documentation/doxygen/Makefile (hunk 2 above)
Comment (etejedor) [reply documentation/doxygen/Makefile:47]: Well in the previous version `$(PYZ_DIR)` was both the input and the output directory, but now you created a new output directory, so it makes sense the python scripts write there directly. @guitargeek can perhaps comment too

Comment (couet) [reply documentation/doxygen/Makefile:47]: Yes and it has also the advantages to avoid the root sources polluting (there was left over)

**Comment by couet** [comment]
I made the changes required by Enric. Now the script takes two input parameters and generates the `.pyzdoc` files directly in the right place. No need to move them afterwards, Can you approve please ?

File: documentation/doxygen/extract_docstrings.py [hunk 4]
```diff
@@ -64,6 +64,8 @@ def run_fast_scandir(dir, ext):    # dir: str, ext: list
     module = ast.parse(file_contents)
     ds = ast.get_docstring(module)
     if ds is not None:
-        with open(pyz_file_path + '.pyzdoc', 'w') as pyz_doc_file:
+        pyz_filename = pyz_file_path
+        pyz_filename = pyz_dir_out + '/' + pyz_filename[pyz_filename.rfind("_") : : ] + 'zdoc'
+        with open(pyz_filename, 'w') as pyz_doc_file:
```
Comment (etejedor) [inline documentation/doxygen/extract_docstrings.py:69]:
```suggestion
        pyz_filename_out = pyz_dir_out + '/' + pyz_file_path[pyz_file_path.rfind("_"):] + 'zdoc'
        with open(pyz_filename_out, 'w') as pyz_doc_file:
```

Comment (etejedor) [reply documentation/doxygen/extract_docstrings.py:69]: In addition: there is still some comment missing to explain the new concatenation of `zdoc`.

**Comment by couet** [comment]
I took your change

**Review by guitargeek — COMMENTED** [review]
Maybe @etejedor has more comments, but for me the changes are fine! It doesn't matter where these pyzdoc files are going, and if by putting them in another directory we can fix the problem with the pyROOT box position I'm good.

Still, I'll also not approve the PR because I think Enric should have the last call.

**Comment by couet** [comment]
I added extra comments. I guess it is ready to be approved.

**Review by etejedor — APPROVED** [review]
